code/DDSCodeTester.cpp

```diff
@@ -7684,6 +7685,71 @@ void pubsub_api_example_create_entities()
     //!--
 }
 
+void statistics_defined_interface()
+{
+    using namespace eprosima::fastdds::rtps;
+    using namespace eprosima::fastdds::dds;
+
+    // STATISTICS_DEDICATED_INTERFACE
+    // --- Device 1 ---
+
+    DomainParticipantQos pqos1;
+
+    // Set NetmaskFilter = ON to enable communication only between locators on the same subnetwork
+    pqos1.transport().netmask_filter = NetmaskFilterKind::ON;
+    
+    // Set ignore_non_matching_locators (Required only in the participant, if no allowlist or blocklist is defined)
+    pqos1.wire_protocol().ignore_non_matching_locators = true;
+
+    // Create DomainParticipant 1
+    DomainParticipant* participant_1 = DomainParticipantFactory::get_instance()->create_participant(0, pqos1);
+    Subscriber* subscriber_1 = participant_1->create_subscriber(SUBSCRIBER_QOS_DEFAULT);
+
+    // --- Device 2 ---
+
+    DomainParticipantQos pqos2;
+
+    // Activate Statistics module
+    pqos2.properties().properties().emplace_back("fastdds.statistics", "DISCOVERY_TOPIC");
+
+    // Set NetmaskFilter = ON to enable communication only between locators on the same subnetwork
+    pqos2.transport().netmask_filter = NetmaskFilterKind::ON;
+
+    // Set ignore_non_matching_locators (Required only in the participant, if no allowlist or blocklist is defined)
+    pqos2.wire_protocol().ignore_non_matching_locators = true;
+
+    // Create DomainParticipant 2
+    DomainParticipant* participant_2 = DomainParticipantFactory::get_instance()->create_participant(0, pqos2);
+
+    // Retrieve Statistics DomainParticipant child from DomainParticipant and create Publisher
+    eprosima::fastdds::statistics::dds::DomainParticipant* statistics_participant =
+            eprosima::fastdds::statistics::dds::DomainParticipant::narrow(participant_2);
+    Publisher* publisher = statistics_participant->create_publisher(PUBLISHER_QOS_DEFAULT);
+
+    // --- Device 3 ---
+
+    // Create a DomainParticipant to monitor statistics information
+    DomainParticipant* monitor_participant = DomainParticipantFactory::get_instance()->create_participant(0,
+                    PARTICIPANT_QOS_DEFAULT);
+    Subscriber* monitor_subscriber = monitor_participant->create_subscriber(SUBSCRIBER_QOS_DEFAULT);
+    DataReaderQos datareader_qos = eprosima::fastdds::statistics::dds::STATISTICS_DATAREADER_QOS;
+
+    // Register the type and create the topic
+    TypeSupport discovery_type;
+    monitor_participant->register_type(discovery_type);
```

Comment:
This compiles but it is not semantically correct, we'd need to instantiate a specific type support